### ISSUE TITLE
Android 12: slow at startup (22 maps 30 sec)

### ISSUE THREAD

**Issue by Rowin63** [osmandapp/OsmAnd#13780]
### Description
When Osmand is started, there is about one minute an empty screen until the map is shown and Osmand is ready to use. 
It seems as this issue came with the update to Android 12, but I´m not quite sure.

### Your Environment
OsmAnd Version: Osmand+ 4.1.11, Storage is Multiuser 1
Android/iOS version: 12
Device model: Galaxy S20+

**Comment by Lee-Carre** [comment]
What I'm the way of testing have you done?
* rebooted the device
* undo any recent changes to OsmAnd
* a fresh installation of OsmAnd
* how is OsmAnd configured? What's the source of the maps you're using (offline vector, online raster)? If offline, are they recently updated (else, what's their date)?
* is your OS fully updated (not just major releases, but recent patches)?
* if you have other devices, do you experience the same behaviour on those? (& how are they different / similar to your S20+)
* does it always happen, every time, or only sometimes?
* do you experience the same problem with other apps since installing Android v12, or only with OsmAnd?

Providing a copy of the log output (under plugins, OsmAnd Development, the entry which mentions logcat), captured just after experiencing this problem, would probably help, too.

**Comment by Rowin63** [comment]
> What I'm the way of testing have you done?
> 
>     * rebooted the device 
done, more than one time.
> 
>     * undo any recent changes to OsmAnd
Dont know what to "undo"- I just use Osmand. 
> 
>     * a fresh installation of OsmAnd
No, Osmand is running fine on two other devices, with mainly the same settings and same data - but Android 11
> 
>     * how is OsmAnd configured? What's the source of the maps you're using (offline vector, online raster)? If offline, are they recently updated (else, what's their date)?
Offline. Always up to date.
> 
>     * is your OS fully updated (not just major releases, but recent patches)?
Yes it is.
> 
>     * if you have other devices, do you experience the same behaviour on those? (& how are they different / similar to your S20+)
Behavour on two other devices is normal.
> 
>     * does it always happen, every time, or only sometimes?
Always, on every Osmand startup
> 
>     * do you experience the same problem with other apps since installing Android v12, or only with OsmAnd?
No. Only with Osmand. But no other app needs to read Osmands map data. 
> 
> 
> Providing a copy of the log output (under plugins, OsmAnd Development, the entry which mentions logcat), captured just after experiencing this problem, would probably help, too.
As I dont know if this file contains any private data, I dont want to post it here in public.

But this line:
02-20 20:44:13.033 31182 31190 I net.osmand.plu: Background concurrent copying GC freed 870780(27MB) AllocSpace objects, 0(0B) LOS objects, 16% free, 121MB/145MB, paused 161us,16us total 217.563ms
appears 22 times in ~30 seconds. 
The end of this lines/commands is the moment, when osmand runs fine again.

**Comment by vshcherb** [comment]
Please try nightly build https://download.osmand.net/latest-night-build/OsmAnd-default.apk, so we can say if it's fixed already or not.

**Comment by vshcherb** [comment]
How many maps do you have ? How much time does it take?

**Comment by Rowin63** [comment]
25 countries about. Bhe number of maps did not change much in the last time (before/after update toAndroid 12). I´m always around 25 countries. And it is noproblem on the other two devices (Android 11).
I tried deleting almost all maps/contour lines/wikipedia, kept only a few (3 or 4) countries - but it made no difference?

In the evening I´ll try the nightly and see. 
Thanks!

**Comment by Rowin63** [comment]
Installed the nightly in the preset "external memory 1".
Because of limited downloads I copied all my 27 maps in its folder and it runs fine!
Also after moving its data to "multiuser 1" (where "my" osmands data is stored)  ist fine.

After this I moved "my" osmand also to "external memory 1" - and it runs fine too...
So at last I selected back to "multiuser 1" for "my" osmand and after moving all its data back to this folder it runs fine too... Very strange. Now everything is like in the beginning, but it runs as expected...

The only difference was, I switched off the Microsoft Earth overlay before changing the data folder (copying these files takes very long). But also after reactivating this overlay at the end again my osmand runs smooth now.
So now everything is like before - but working.
